lib/v2/screens/transfer/send/send_confirmation/interactor/usecases/send_transaction_use_case.dart

```diff
@@ -10,14 +11,26 @@ class SendTransactionUseCase {
   final ProfileRepository _profileRepository = ProfileRepository();
   final fromAccount = settingsStorage.accountName;
 
-  Future<Result> run(String? toName, String account, Map<String, dynamic> data) {
-    return _sendTransactionRepository.sendTransaction(toName, account, data, fromAccount).then((Result value) async {
+  Future<Result> run({required String actionName, required String account, required Map<String, dynamic> data}) {
+    return _sendTransactionRepository
+        .sendTransaction(actionName: actionName, account: account, data: data, accountName: fromAccount)
+        .then((Result value) async {
       if (value.isError) {
         return value;
       } else {
-        final List<Result> profiles = await getProfileData(data['to'], fromAccount);
-        final transactionModel = TransactionModel.fromTxData(data, value.asValue!.value);
-        return ValueResult(SendTransactionResponse(profiles, value, transactionModel));
+        final String transactionId = value.asValue!.value;
+        final transactionModel = GenericTransactionModel(
+            account: account,
+            action: actionName,
+            data: data,
+            transactionId: transactionId,
+            timestamp: DateTime.now().toUtc());
+        final transferModel = TransactionModel.fromTransaction(transactionModel);
+        List<Result> profiles = [];
+        if (transferModel != null) {
+          profiles = await getProfileData(transferModel.to, fromAccount);
```

Comment:
In the old code, it would only handle transfer, and load profile data of the involved parties. 

We still want to load profile data in case of a transfer, but not for other transactions